### PR TITLE
build: migrate ts-loader to babel-loader

### PR DESCRIPTION
**Pull request by nataliepina**
Closes D2IQ-92433

# Description
By default, Storybook is configured to work with Babel - https://storybook.js.org/docs/react/configure/babel
Babel’s ecosystem has evolved for supporting typescript in recent years. Benefits of fully migrating to Babel include better tree-shaking resulting in smaller bundle size compared to using ts-loader with parts of babel. Babel is overall faster than ts-loader. In many Storybook Webpack 5 migration guides, I am seeing babel used instead of ts-loader. So I've decided to try out the migration here in the hopes of decreasing our bundle size. I've also included a script to run for checking types.

By default, babel-loader will store the cache data in node_modules/. cache/babel-loader.

I believe the renovate.json file is a remnant of the Jenkins configuration as it has much older versions of dependencies and the commit history points to 4 years ago. 

## Which issue(s) does this PR relate to?
https://jira.d2iq.com/browse/D2IQ-92433

## Testing
Break a prop type and run the new types test:
`npm run test:types`

## Trade-offs
There are a few things we cannot do until we update Typescript to v4+ such as including `--noEmit` in the types check. 

## Checklist

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [x] This PR is associated with a JIRA and mentions in the commit message footer ("Closes …")
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [x] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -17,7 +16,10 @@
     "moduleResolution": "node",
     "skipLibCheck": true,
     "tsBuildInfoFile": ".tsbuildinfo",
-    "lib": ["es6", "dom", "es2017"]
+    "lib": ["es6", "dom", "es2017"],
+    "incremental": true,
```
Comment (mstruebing) [inline tsconfig.json:20]: This is such a small but great improvement! 👍

File: package.json [hunk 2]
```diff
@@ -36,7 +36,8 @@
     "test": "npx jest",
     "test-storybook": "test-storybook",
     "chromatic": "chromatic",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "test:types": "tsc -p tsconfig.json"
```
Comment (mstruebing) [inline package.json:40]: Do we want to add this to our `pre-commit` hooks and Github-Actions?

Comment (nataliepina) [reply package.json:40]: Yes, we should! I'll include it here. 🎉

File: tsconfig.json (hunk 1 above)
Comment (seialkali) [reply tsconfig.json:20]: Could I ask what these changes are doing? I'm not very familiar with tsconfig options.

Comment (mstruebing) [reply tsconfig.json:20]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#faster-subsequent-builds-with-the---incremental-flag

It makes things faster.

Comment (seialkali) [reply tsconfig.json:20]: Ahh ok, it was already in the config, just moved to this line. Does the placement in the config make a difference?

Comment (nataliepina) [reply tsconfig.json:20]: The placement does not make a difference here.

**Review by mstruebing — APPROVED** [review]

**Pull request merged**